File: python/functions/combine_arrays.py
"""
Combines multiple 3-D distortion maps by taking the element-wise maximum
(ignoring NaN values where any map has data).
"""

import numpy as np


def combine_arrays(*arrays: np.ndarray) -> np.ndarray:
    """
    Parameters
    ----------
    *arrays : one or more np.ndarray of identical shape

    Returns
    -------
    np.ndarray
        Element-wise NaN-aware maximum across all input arrays.
    """
    if len(arrays) == 0:
        raise ValueError('At least one input required.')

    stacked = np.stack([a for a in arrays], axis=-1)  # (..., n)
    return np.nanmax(stacked, axis=-1)

File: matlab/functions/combineArrays.m
function combined_dis_map = combineArrays(varargin)
    % Check inputs
    n = numel(varargin);
    if n == 0
        error('At least one input required.');
    end
    
    arrays = cellfun(@(x) x(:,:,:), varargin, 'UniformOutput', false);
    
    % Stack and average
    combined_dis_map = cat(4, arrays{:});
    combined_dis_map = max(combined_dis_map,[],4,"omitnan");
    
end
